api/src/test/java/org/openmrs/module/queue/api/QueueEntryServiceTest.java

```diff
@@ -109,6 +109,48 @@ public void shouldCreateNewQueueEntryRecord() {
 		assertThat(result.getPriority(), is(conceptPriority));
 	}
 	
+	@Test
+	public void shouldNotCreateDuplicateOverlappingQueueEntryRecords() {
+		Queue queue = new Queue();
```

Comment:
@cioan - note, I removed all of the mocking in favor of just creating and populating the actual objects, they are just beans.